### PR TITLE
Fix empty bag equals null bag

### PR DESCRIPTION
**Pull request by julienledem**

### CODE REVIEW

File: redelm-column/src/main/java/redelm/Log.java [hunk 1]
```diff
@@ -31,6 +31,10 @@
  * Adds compile time log level.
  * The compiler removes completely if statements that reference to a false constant
  *
+ * <code>
+ *   if (DEBUG) LOG.debug("removed by the compiler if DEBUG is a false constant")
+ * <code>
```
Comment (jcoveney) [inline redelm-column/src/main/java/redelm/Log.java]: the final code tag needs a backslash

File: redelm-pig/src/main/java/redelm/pig/PigSchemaConverter.java [hunk 2]
```diff
@@ -33,10 +33,9 @@
  *
  * Converts a Pig Schema into a RedElm schema
  *
- * Bags are converted into repeated fields
- * Map are converted into repeated groups of key/values
- *
- * TODO: add an optional group containing the repeated field for bags so that null != empty
+ * Bags are converted into an optional group containing one repeated group field to preserve distinction between empty bag and null.
+ * Map are converted into an optional group containing one repeated group field of (key, value).
+ * anonymous fields are named field_{index}. (apparently pig already gives them an alias val_{int}, so this never happens)
```
Comment (jcoveney) [inline redelm-pig/src/main/java/redelm/pig/PigSchemaConverter.java]: In most cases pig will give them an alias, but in some cases it won't. The most common example is if a UDF doesn't give it a name (which I could change, but is how it is currently)

File: redelm-pig/src/test/java/redelm/pig/TestPigSchemaConverter.java [hunk 3]
```diff
@@ -26,20 +26,69 @@
 
 public class TestPigSchemaConverter {
```
Comment (jcoveney) [inline redelm-pig/src/test/java/redelm/pig/TestPigSchemaConverter.java:27]: I think that testing that the structure is what we think it should be is good. I also think we need tests that make sure that it works. IE now we can have a null bag OR an empty (but not null) bag. We should test this.

Comment (julienledem) [inline redelm-pig/src/test/java/redelm/pig/TestPigSchemaConverter.java:27]: I added more tests in a separate branch

**Pull request merged**